components/ca-crud.html

```diff
@@ -687,12 +687,24 @@
 
             // extracts action information to be used by the UI such as button labels
             actions.forEach(item => {
-
-                // store action using its name as the key, this allows retrive
+                // store action using its name as the key, this allows retrieve
                 // action based on button label clicked from dialog
                 labelToAction[item.title] = item;
             });
 
+            // get a list of navigations from hateoas links with URN nav:name
+            const navs = schemaActions.getNavsFromData(data);
+
+            // extracts navigation information to be used by the UI such as button labels
+            const labelToNav = {};
+
+            // extracts navigation information to be used by the UI such as button labels
+            navs.forEach(item => {
+                // store navigation using its name as the key, this allows retrieve
+                // navigation based on button label clicked from dialog
+                labelToNav[item.title] = item;
+            });
```

Comment:
If you want to avoid updating an external object, this can be expressed as
```
const labelToNav = reduce((labels, item) => { 
  labels[item.title] = item; 
  return labels; 
}, {})
```